src/main/java/org/takes/rq/RqBuffered.java

```diff
@@ -45,17 +43,10 @@
      */
     public RqBuffered(final Request req) {
         super(
-            new Request() {
-                @Override
-                public Iterable<String> head() throws IOException {
-                    return req.head();
-                }
-
-                @Override
-                public InputStream body() throws IOException {
-                    return new BufferedInputStream(req.body());
-                }
-            }
+            new RequestOf(
+                req::head,
+                () -> new BufferedInputStream(req.body())
```

Comment:
@fanifieiev use the secondary constructor.

Reply:
@fabriciofx I cannot use the secondary constructor here.
![image](https://user-images.githubusercontent.com/3167940/69067336-e363f000-09f0-11ea-954d-455caf102a4d.png)

Reply:
@fanifieiev as I said [here](https://github.com/yegor256/takes/pull/995#discussion_r347074093): *Of couse you can't use it in all places (e.g. beacause it can throws an Exception). So, in these places you can use the primary constructor...*